Include/internal/pycore_stackref.h

```diff
@@ -219,7 +219,7 @@ PyStackRef_FromPyObjectNew(PyObject *obj)
     // Make sure we don't take an already tagged value.
     assert(((uintptr_t)obj & Py_TAG_BITS) == 0);
     assert(obj != NULL);
-    if (_Py_IsImmortal(obj) || _PyObject_HasDeferredRefcount(obj)) {
+    if (_PyObject_HasDeferredRefcount(obj)) {
```

Comment:
There are two more places where we do `_Py_IsImmortal(obj) || _PyObject_HasDeferredRefcount(obj)` that we might as well combine:

* `pycore_object.h`: _Py_TryIncrefCompareStackRef
* `dictobject.c`: _Py_dict_lookup_threadsafe_stackref